### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Updated ReadMe.md.

**Related issue(s)**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,48 +3,124 @@ Manages the lifecycle of a NATS JetStream deployment.
...
+**NOTE:** Run `make --help` for more information on all potential `make` targets
+
+More information can be found via the [Kubebuilder Documentation](https://book.kubebuilder.io/introduction.html)
+
+### Build container images
+
+Build and push your image to the location specified by `IMG`:
 
 ```sh
 make docker-build docker-push IMG=<some-registry>/nats-manager:tag
```
Comment (grischperl) [inline README.md:72]: `tag` is supposed to be the (version) tag you give the image, right? If you're supposed to put a value there yourself, I think markers like `<tag>` would be nice

Comment (grischperl) [inline README.md:72]: This is optional, but a hint for the value you're supposed to put in `<some-registry>` would be nice. Like _Note: If using docker, it is usually your username_.

File: README.md [hunk 2]
```diff
@@ -3,48 +3,124 @@ Manages the lifecycle of a NATS JetStream deployment.
...
-**Note:** Your controller will automatically use the current context in your kubeconfig file (i.e. whatever cluster `kubectl cluster-info` shows).
 
-### Running on the cluster
-1. Install Instances of Custom Resources:
+### How it works
+
+This project aims to follow the Kubernetes [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/).
+
+It uses [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/),
+which provide a reconcile function responsible for synchronizing resources until the desired state is reached on the cluster.
```
Comment (grischperl) [inline README.md:12]: Maybe another sentence here, that it's based on Kubebuilder which offers commands and `make` targets which create code based on markers. And that you should refer to these when modifying stuff.

File: README.md (hunk 1 above)
Comment (mfaizanse) [reply README.md]: updated

Comment (mfaizanse) [reply README.md]: added a comment next to the command.

**Review by grischperl — APPROVED** [review]

**Pull request merged**